enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,253 @@
+    of all nodes to the address set for that namespace, causing
+    node-to-pod and pod-to-node traffic to be affected by
+    NetworkPolicies that match that namespace.
+
+    (FIXME: would that actually work? Might need further tweaks...)
+
+Given that change to upstream ovn-kubernetes we can then configure
+ovn-kubernetes-using OCP clusters as follows:
+
+  1. We set `host-network-namespace=openshift-host-network` in the
+     ovn-kubernetes config
+
+  2. CNO creates an `openshift-host-network` namespace, with the label
+     `network.openshift.io/policy-group: host-network`
+
+And for openshift-sdn clusters:
+
+  1. CNO ensures that the `default` namespace has the label
+     `network.openshift.io/policy-group: host-network`
+
+Then users will be able to match router traffic in either
+openshift-sdn or ovn-kubernetes with any router endpoint publishing
+strategy via:
+
+    ingress:
+      - from:
+          - namespaceSelector:
+              matchLabels:
+                network.openshift.io/policy-group: router
+          - namespaceSelector:
+              matchLabels:
+                network.openshift.io/policy-group: host-network
+
+### Alternatives
+
+- (We don't want ovn-kubernetes to apply the magic behavior to a
+  _fixed_ namespace name, because in OpenShift we will need the magic
+  namespace to have an `openshift`-prefixed name, but we can't force
+  that on upstream ovn-kubernetes.)
+
+- Instead of specifying a namespace name in the configuration, we
+  could say that the behavior is applied to the namespace that has a
+  particular annotation, eg `k8s.ovn.org/host-network-namespace`. But
+  then there's the question of what to do if there are multiple
+  namespaces with that annotation, so the config option seems better.
+
+- An alternative possibility is to not have an _actual_
+  `openshift-host-network` namespace, but to instead just have a
+  particular magic/virtual label. (Eg, in OCP we would specify
+  `host-network-selector="network.openshift.io/policy-group:
+  host-network"` in the ovn-kubernetes config, and then ovn-kubernetes
+  makes it so that any NetworkPolicy containing that selector ends up
+  applying to host-network traffic.) But this seems like it would be
+  much more complicated to implement in ovn-kubernetes. (We would
+  probably end up just creating a fake internal namespace anyway.)
+
+- I don't think we should use `default` as the magic namespace in
+  ovn-kubernetes, because `default` isn't actually supposed to be
+  magic in any way; that's just a weird quirk of openshift-sdn.
+
+- On the openshift-sdn side, rather than labeling `default`, we could
+  instead add special handling for `openshift-host-network` (ie,
+  treating it as VNID 0 for purposes of NetworkPolicy regardless of
+  its actual VNID) and then just have CNO create and label the
+  `openshift-host-network` namespace regardless of network plugin.
+
+- Instead of recommending that people explicitly allow traffic from
```

Comment:
This would be my preferred option. `policy-group: router` is the "published" API.

Reply:
I guess maybe instead of having cluster-ingress-operator do the labeling, CNO could do it; then CNO would be consuming a documented CIO API (its config object) rather than CIO consuming an undocumented CNO API (the name of the magic-host-network namespace).